File: scripts/python/main_pipeline.py
import subprocess

scripts = [
    "scripts/python/data_generation.py",
    "scripts/python/model_training.py",
    "scripts/python/output_analysis.py"
]

for script in scripts:
    try:
        print(f"Running {script}")
        subprocess.run(["python3", script], check=True)
    except subprocess.CalledProcessError as e:
        print(f"Error while running {script}: {e}")
        exit(1)
print("Pipeline complete.")

File: scripts/r/main_pipeline.R
scripts <- c("scripts/data_generation.R",
             "scripts/model_training.R",
             "scripts/output_analysis.R")

for (script in scripts) {
  cat("Running", script, "...\n")
  tryCatch(
    { source(script) },
    error = function(e) {
      cat("Error in", script, ":\n", e$message, "\n")
      stop("Aborting pipeline.")
    }
  )
}
cat("Pipeline completed successfully.\n")
